### PR TITLE
Add extra makeshift magazines

### PR DESCRIPTION
**Pull request by Rail-Runner**
#### Summary

```SUMMARY: Content "Add extra makeshift magazines"```
#### Purpose of change

Add extra makeshift magazines for .308, .223 and .45 guns that could have them. Gun which could have a magazine improvised for it must not use angled-feed magazines and must use a box magazine; all makeshift magazines are straight single-stack box type. Also convert .308 pipe rifle to use a detachable magazine rather than a tube magazine, as it was said that feeding from tube magazine isn't something that could be made with the currently available tools. 
#### Describe the solution

JSON editing.
#### Describe alternatives you've considered

None.
#### Testing

Loaded up the game with newly-added magazines and recipes, looked at their stat blocks via debug menu to see if stats are fine.
#### Additional context

None.

### CODE REVIEW

File: data/json/items/magazine/308.json [hunk 1]
```diff
@@ -44,6 +44,23 @@
     "reliability": 9,
     "flags": [ "MAG_COMPACT" ]
   },
+  {
+    "id": "fal_makeshiftmag",
+    "type": "MAGAZINE",
+    "name": "FN FAL makeshift magazine",
+    "description": "An improvised 5-round straight single-stack box magazine consisting of little more than a bent sheet of steel held together by duct tape and hope, for use with FN FAL rifle.",
+    "weight": "80 g",
+    "volume": "250 ml",
```
Comment (LaVeyanFiend) [inline data/json/items/magazine/308.json]: I don't think magazines just so happen to have volumes based on the old 250ml volume units 😛

Comment (Rail-Runner) [reply data/json/items/magazine/308.json]: I can't think of any way to better estimate the volume, especially considering that this is a makeshift magazine and all stats would actually vary wildly depending on the maker.

Comment (LaVeyanFiend) [reply data/json/items/magazine/308.json]: @tenmillimaster & I(Tenmilli is more experienced at it) have both used [ImageJ](https://imagej.nih.gov/ij/) on images of guns to find volume, technique of which is easily applicable to magazines(use gun with a known length, scale image based on length, use scale on magazine to find its volume as well). I'd make its volume equal to the magazine it's copying this way(with the lower capacity as a penalty for being makeshift, obv).

Comment (Rail-Runner) [reply data/json/items/magazine/308.json]: Actually the makeshift magazine would have to have less volume, I think. It'd be thinner due to being single-stack (most copied magazines are double-stack), and also shorter, and you can't really make larger magazines due to them becoming way too unreliable to be at least somewhat usable.

**Pull request merged**